core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -753,14 +627,15 @@ private RecordCursorFactory generateJoins(QueryModel model, SqlExecutionContext
                                     );
                                 }
                             } else {
-                                master = createFullFatAsOfJoin(
+                                master = createFullFatJoin(
                                         master,
                                         masterMetadata,
                                         masterAlias,
                                         slave,
                                         slaveMetadata,
                                         slaveModel.getName(),
-                                        slaveModel.getJoinKeywordPosition()
+                                        slaveModel.getJoinKeywordPosition(),
+                                        SqlCodeGenerator::createFullFatAsOfJoin
```

Comment:
could we make this a static final variable? Annoyingly this is an allocating call in java